services/graphite/parser_test.go

```diff
@@ -0,0 +1,513 @@
+package graphite_test
+
+import (
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/influxdb/influxdb/services/graphite"
+	"github.com/influxdb/influxdb/tsdb"
+)
+
+func BenchmarkParse(b *testing.B) {
+	p, err := graphite.NewParser([]string{
+		"*.* .wrong.measurement*",
+		"servers.* .host.measurement*",
+		"servers.localhost .host.measurement*",
+		"*.localhost .host.measurement*",
+		"*.*.cpu .host.measurement*",
+		"a.b.c .host.measurement*",
+		"influxd.*.foo .host.measurement*",
+		"prod.*.mem .host.measurement*",
+	}, nil)
+
+	if err != nil {
+		b.Fatalf("unexpected error creating parser, got %v", err)
+	}
+
+	for i := 0; i < b.N; i++ {
+		p.Parse("servers.localhost.cpu.load 11 1435077219")
+	}
+}
+
+func TestTemplateApply(t *testing.T) {
+	var tests = []struct {
+		test        string
+		str         string
+		measurement string
+		tags        map[string]string
+		template    string
+		err         string
+	}{
+		{test: "metric only",
```

Comment:
Minor observation about this test data. I find it easier if the inputs to the test -- `str` and `template` are grouped first, the expected output comes next. Or perhaps prefix expected data with `exp` e.g. `expMeasurement`. Just a style thing, but I had to study the test code to understand what goes in, and what should come out.